packages/driver/src/cypress/error_utils.js

```diff
@@ -168,174 +166,169 @@ const warnByPath = (errPath, options = {}) => {
   $utils.warning(err)
 }
 
+class InternalCypressError extends Error {
+  constructor (message) {
+    super(message)
+
+    this.name = 'InternalCypressError'
+
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, InternalCypressError)
+    }
+  }
+}
+
+class CypressError extends Error {
+  constructor (message) {
+    super(message)
+
+    this.name = 'CypressError'
+
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, CypressError)
+    }
+  }
+}
+
 const internalErr = (err) => {
-  const newErr = new Error(err)
+  const newErr = new InternalCypressError(err.message)
 
-  return mergeErrProps(newErr, err, { name: 'InternalError' })
+  return mergeErrProps(newErr, err)
 }
 
 const cypressErr = (err) => {
-  const newErr = new Error(err.message)
+  const newErr = new CypressError(err.message)
 
-  return mergeErrProps(newErr, err, { name: 'CypressError' })
+  return mergeErrProps(newErr, err)
 }
 
 const cypressErrByPath = (errPath, options = {}) => {
-  const errObj = errObjByPath($errorMessages, errPath, options.args)
+  const errObj = errByPath(errPath, options.args)
 
   return cypressErr(errObj)
 }
 
-const normalizeMsgNewLines = (message) => {
-  //# normalize more than 2 new lines
-  //# into only exactly 2 new lines
-  return _
-  .chain(message)
-  .split(twoOrMoreNewLinesRe)
-  .compact()
-  .join('\n\n')
-  .value()
-}
-
 const replaceErrMsgTokens = (errMessage, args) => {
   if (!errMessage) return errMessage
 
+  const replace = (str, argValue, argKey) => {
+    return str.replace(new RegExp(`\{\{${argKey}\}\}`, 'g'), argValue)
+  }
+
   const getMsg = function (args = {}) {
     return _.reduce(args, (message, argValue, argKey) => {
-      return message.replace(new RegExp(`\{\{${argKey}\}\}`, 'g'), argValue)
+      if (_.isArray(message)) {
+        return _.map(message, (str) => replace(str, argValue, argKey))
+      }
+
+      return replace(message, argValue, argKey)
```

Comment:
You can normalize this and DRY it up a bit more like this...

```js
const replace = (argValue, argKey) => {
  return (str) => {
    return str.replace(new RegExp(`\{\{${argKey}\}\}`, 'g'), argValue)
  }
}

return _
.chain(message)
.concat([]) // normalize into array
.map(replace(argValue, argKey))
.thru((val) => {
  if (_.isArray(message)) {
    return val[0]
  }
  
  return val
})
.value()
```

Reply:
I don't think this does the same thing. It needs to return a string if `message` is a string and return an array if it's an array. This seems to always return an array and only include the first value if it is an array?